blocksync/buffer.go

```diff
@@ -37,47 +37,45 @@ type blockBuffer struct {
 func (b *blockBuffer) Flush(blk *blockchain.Block) (bool, bCheckinResult) {
 	b.mu.Lock()
 	defer b.mu.Unlock()
-	moved := false
 	if blk == nil {
-		return moved, bCheckinSkipNil
+		return false, bCheckinSkipNil
 	}
 	confirmedHeight := b.bc.TipHeight()
 	// check
 	h := blk.Height()
 	if h <= confirmedHeight {
-		return moved, bCheckinLower
+		return false, bCheckinLower
 	}
 	if _, ok := b.blocks[h]; ok {
-		return moved, bCheckinExisting
+		return false, bCheckinExisting
 	}
 	if h > confirmedHeight+b.size {
-		return moved, bCheckinHigher
+		return false, bCheckinHigher
 	}
 	b.blocks[h] = blk
 	l := logger.With().
 		Uint64("recvHeight", blk.Height()).
 		Uint64("confirmedHeight", confirmedHeight).
 		Str("source", "blockBuffer").
 		Logger()
-	for b.size > 0 {
-		next := confirmedHeight + 1
-		blk, ok := b.blocks[next]
+	syncedHeight := confirmedHeight
+	for confirmedHeight+b.size > syncedHeight {
```

Comment:
`for syncedHeight := confirmedHeight; confirmedHeight+b.size > syncedHeight;syncedHeight++`